packages/carbon-component-mapper/src/text-field/text-field.js

```diff
@@ -11,6 +11,10 @@ const TextField = (props) => {
 
   const Component = input.type === 'number' ? NumberInput : TextInput;
 
+  const setValue = (e, input) => {
+    input.value = input.type === 'number' ? e.imaginaryTarget.value : e.target.value;
```

Comment:
This won't work. `input` is immutable, you cannot rewrite. You have to call `onChange` to change it and trigger a new render.

Reply:
@rvsia this code is working in my local, does this work?
```
 const setValue = (e, input) => {
    if (input.type === 'number') {
      return input.onChange(e.imaginaryTarget.value);
    } else {
      return input.onChange(e.target.value);
    }
  };

Reply:
@kavyanekkalapu it "works" because there are also other events triggering that are causing the re-render. If you start limiting the `subscription` prop to the field, it will stop working at some point.
The `onChange` function is expecting either an HTML event or a value not undefined.

The code snippet you posted should work. Although this would be closer to our code style (forgive any typos in code example I am writing this on my phone)

```jsx
const getValue = (event) => event.type === 'number' ? event.imaginaryTarget.value : event.target.value

...

<Component
  ...
  onChange={event => input.onChange(getValue(event))}
  {...(input.type === 'number' ? { label: labelText } : { labelText })}
  {...rest}
/>

```

I would also like to see a test case.